sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/DLPDeidentifyText.java

```diff
@@ -177,19 +177,24 @@ public DLPDeidentifyText build() {
   @Override
   public PCollection<KV<String, DeidentifyContentResponse>> expand(
       PCollection<KV<String, String>> input) {
-    return input
-        .apply(ParDo.of(new MapStringToDlpRow(getColumnDelimiter())))
-        .apply("Batch Contents", ParDo.of(new BatchRequestForDLP(getBatchSizeBytes())))
-        .apply(
-            "DLPDeidentify",
+
+    ParDo.SingleOutput<KV<String, Iterable<Table.Row>>, KV<String, DeidentifyContentResponse>>
+        deidentifyParDo =
             ParDo.of(
                 new DeidentifyText(
                     getProjectId(),
                     getInspectTemplateName(),
                     getDeidentifyTemplateName(),
                     getInspectConfig(),
                     getDeidentifyConfig(),
-                    getHeaderColumns())));
+                    getHeaderColumns()));
+    if (getHeaderColumns() != null) {
```

Comment:
Do we need to add new unit tests to cover the new functionality ?

Reply:
It would be nice to add them, but I have no idea how to approach checking this. Do you have any suggestions?

Reply:
I'm not sure of all the available tools for testing DoFns in Beam so there may be better options but one way would be to use the @NeedsRunner annotation and some PAsserts to verify the expected output ([from the wiki](https://cwiki.apache.org/confluence/display/BEAM/Contribution+Testing+Guide#ContributionTestingGuide-EffectiveuseoftheTestPipelineJUnitrule))? 

It would require injecting a mock cloud client library to avoid making any calls to the live API.

Reply:
It was just a suggestion. I'm OK with merging as it is if that's OK with the author of the PR.

Reply:
Thanks, merging for now so that we don't have to cherry-pick this into 2.24.0 and we have manually verified the fix on 2.23.0.